tests/test_AggregateGroupData.cpp

```diff
@@ -0,0 +1,589 @@
+/*
+  Copyright 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define BOOST_TEST_MODULE Aggregate_Group_Data
+#include <opm/output/eclipse/AggregateGroupData.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/output/eclipse/AggregateWellData.hpp>
+
+#include <opm/output/eclipse/VectorItems/intehead.hpp>
+#include <opm/output/eclipse/VectorItems/well.hpp>
+
+#include <opm/output/data/Wells.hpp>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+
+#include <exception>
+#include <stdexcept>
+#include <utility>
+#include <vector>
+#include <iostream>
+#include <cstddef>
+
+struct MockIH
+{
+    MockIH(const int numWells,
+	   
+	   const int igrpPerGrp	 = 101,  // no of data elements per group in IGRP array
```

Comment:
These numbers - are they specific to this testcase?

Reply:
No - the numbers that are assigned constant values, always has these values. I.e. igrpPerGrp, sgrpPerGrp, xgrpPerGrp and zgrpPerGrp.

Reply:
OK - sorry for being a bit dense; what I am after is just that *iff* these numbers are truly constant they should be defined as named constants in the output code, and not inline in the test code. But maybe I misunderstand.